iothub/device/src/Transport/Amqp/AmqpUnit.cs

```diff
@@ -665,9 +655,10 @@ public async Task<AmqpIotOutcome> SendEventAsync(Message message, CancellationTo
             }
         }
 
-        public void OnEventsReceived(Message message)
+        public Task OnEventsReceived(Message message)
         {
             _onMessageReceivedCallback?.Invoke(message);
+            return Task.CompletedTask;
```

Comment:
Why are we returning a generic completed task? If the caller expects a task in return, then shouldn't that be the invoked task?

Reply:
I've fixed this in my latest commit to this PR.